### DISCUSSION TITLE
wire:click is not working if it wasn't rendered on the first run

### DISCUSSION THREAD

**Discussion by ChrisSantiago82** [livewire/livewire#1861] [1. Help] [answered]
I have an application where a user can select products. There is a limit on how many products the user can select. I have a button to select the products that is connected with wire:click. If the limit is reached I'm showing a gray button and this button doesn't has a wire:click.
So, there is a if else that shows either the button with the wire:click or the other button. Everything works perfectly. If the limit is reached, the gray buttons are showed, if something is removed the other buttons appear again.
Now I have the case that the products are preselected and it's already over the limit on the first render. Only when the user removes some products he can select a new product.
On fist render there is no wire:click only later if the condition changes the wire:click button appears. And in this case the button doesn't call the method. There is now error in the console, nothing. If there was never that wiere:click on the first render, I can't add it on a later time. If it was there on the first render, everything works perfectly:

`@if ($this->amountOpenToSelect > (($product->price / $product->amount) * $inSelectedAmount[$product->id]))

<button class="selectbtn float-right" wire:click="addProduct('{{ $product->id }}')">Auswählen </button>

@else

<button class="selectbtnDisabled float-right">Auswählen</button>

@endif`

**Context**
- Livewire version: 2.3
- Laravel version: 7
- Browser: Chrome

**Comment by iksaku** [comment]
Could you please try to replicate your issue in a [Laravel Playground](https://laravelplayground.com/#/) please?

**Comment by ChrisSantiago82** [comment]
Hey,
I'm new to Laravel Playground, so please be patient if I don't do it the correct way. But I could reproduce the error there:

https://laravelplayground.com/#/snippets/4e2c7ad6-3199-4f8e-83a2-7662c9f76504

First time when it's rendered, the button "Can't clickButton" isn't connected to any method (has no wire click).
If you click the "Enable/Disable Button", that will remove the "Can't clickButton" and add the "Click Button". The click Button has a wire:click, but that method is never being called because it was not there in the first rendering.

Does that make any sense?

**Reply by iksaku** [reply]
Yup, makes sense.

This is one of the most common issues when dealing with different HTML elements, but using the same element type.

In order to make it work, you need to tell Livewire how to **distinguish** between each element by giving each one a name, using `wire:key`.

[Here's a playground with the `wire:key` fix](https://laravelplayground.com/#/snippets/afc4d8e2-f500-4122-8cde-127e365149c5), for more information, please go to the [Troubleshooting Documentation](https://laravel-livewire.com/docs/2.x/troubleshooting).

**Comment by ChrisSantiago82** [comment, marked as answer]
Ok, I get it. I thought the wire key is only necessary in loops. 
Then I will close this issue since it's my error.
